File: go-algorithm/leetcode/problem151.go
package main

import "strings"

// 151. Reverse Words in a String
// https://leetcode.com/problems/reverse-words-in-a-string/

func reverseWords(s string) string {
	arr := strings.Split(s, " ")
	var store []string
	for _, word := range arr {
		if word != "" {
			store = append(store, word)
		}
	}
	ans := ""
	for i := len(store) - 1; i >= 0; i-- {
		ans += store[i]
		if i > 0 {
			ans += " "
		}
	}
	return ans
}

File: java-algorithm/src/main/java/com/leetcode/Problem151.java
package com.leetcode;

import java.util.ArrayList;
import java.util.List;

/**
 * 151. Reverse Words in a String
 * https://leetcode.com/problems/reverse-words-in-a-string/
 */
public class Problem151 {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        List<String> reversed = new ArrayList<>();
        for (int i = words.length - 1; i >= 0; i--) {
            if (words[i].length() > 0) {
                reversed.add(words[i]);
            }
        }
        return String.join(" ", reversed);
    }
}
